.github/workflows/ci.yaml

```diff
@@ -63,6 +71,9 @@ jobs:
     - id: checkout
       name: Check out source code
       uses: actions/checkout@v3
+      with:
+        ref: ${{github.head_ref}}
+        repository: ${{github.event.pull_request.head.repo.full_name}}
```

Comment:
These checkout actions will run with the code of the head of the PR, they won't run unless the PR is labeled as safe to run though.